CONTRIBUTING.md

```diff
@@ -0,0 +1,152 @@
+# Contributing <!-- omit in toc -->
+
+Thank you for taking the time to contribute to our open-source projects!
+Contributions are welcome and you will be fully credited.
+
+Please take the time to read and understand the contribution guide before
+creating an issue or pull request.
+
+- [Code of Conduct](#code-of-conduct)
+- [Contribution Guidelines](#contribution-guidelines)
+  - [Reporting Issues/Bugs](#reporting-issuesbugs)
+- [Maintainership](#maintainership)
+- [Best Practices](#best-practices)
+  - [Branch Names](#branch-names)
+  - [Coding Style](#coding-style)
+    - [WordPress Coding Standard](#wordpress-coding-standard)
+    - [Tests](#tests)
+    - [Backward Compatibility](#backward-compatibility)
+    - [Documentation](#documentation)
+    - [Compiled Assets](#compiled-assets)
+  - [Release Process](#release-process)
+    - [Releases to WordPress.org](#releases-to-wordpressorg)
+
+## Code of Conduct
+
+Please read our [Code of Conduct](./CODE_OF_CONDUCT.md) to keep our open source
+community approachable and respectable.
+
+## Contribution Guidelines
+
+Contributions to projects can be in the form of bug reports or feature requests
+via issues or pull requests. Pull requests are the preferred method of
+contribution to a project.
+
+### Reporting Issues/Bugs
+
+Issues are the best way to file a bug report for a project. Before filing an
+issue, please check for the following:
+
+- Attempt to replicate your problem to ensure it can be repeated and isn't
+  coincidental.
+- Ensure that your feature request is within the scope of the project. For
+  example, please don't suggest adding Drupal support to a WordPress plugin. If
+  you are unsure, try making a discussion post in the project to ask the project
+  maintainers.
+- Search the repository for an existing issue/pull request to your issue/feature
+  request. Check to make sure that your issue wasn't already reported/requested.
+
+## Maintainership
+
+All repositories should have a [CODEOWNERS](codeowners) file included with a
+specified owner for the project. Preferably a team or 2+ individuals would be
+marked as "owners" who are responsible for the project. These owners are
+responsible for the maintainership of the project and controlling the scope of
+features within the said project. They shall have the final say on whether or not a
+feature request is a good fit for the scope of a project.
+
+## Best Practices
+
+The following are best practices that we strive for in all of our projects. A
+project may override these where it makes sense. The respective project should
+have an updated `CONTRIBUTING.md` included with information about best practices
+for the project.
+
+### Branch Names
+
+The default branch for repositories is `trunk`. Always work from the `trunk`
+branch and open up pull requests against it.
```

Comment:
I thought we discussed `main` in our meeting, with the option to default to a `develop` branch if it's possible that someone would install the repo code directly (e.g., as a submodule), so that `main` would only include the latest stable release?

Reply:
Maybe it's worth clarifying the type of project this will apply to. This doc is framed in the mindset of the 'package' type of project vs 'hosted' ones where the default branch would still be `main`.

I'll switch out the default branch name and clarify the language to share that point a bit more clearly.

Reply:
The updates in [9fa96aa](https://github.com/alleyinteractive/.github/pull/7/commits/9fa96aa4515229540423911e7f155e5200209a81) are better, but not exactly what I had in mind. Generally, I think that package repos should still use `main` as their default branch — **except** in cases where the package would be installed directly from the default branch of the repo, e.g. a plugin that might be installed as a submodule or downloaded as a zip. In those cases, the default branch could be `develop`, indicating that it may be unstable for production. In those cases, the `main` branch should always include the latest released stable version.

However, now that I type that, I don't know that the distinction makes that much sense because `git clone` will checkout the default branch regardless of what it's called, so maybe just sticking with `main` is fine?